reccmp/isledecomp/compare/csv.py

```diff
@@ -0,0 +1,243 @@
+"""Parsing files in csv format.
+The python csv.DictReader class does the heavy lifting.
+We begin with a pre-processing step that removes blank lines or "comment" lines that begin with # or //.
+The delimiter is decided based on the first line (following pre-processing). Options are: pipe, comma, tab
+"""
+
+import csv
+from csv import Error as PythonCsvError
+from typing import Iterable, Iterator
+from typing_extensions import NotRequired, TypedDict
+from reccmp.isledecomp.types import EntityType
+
+
+# Fatal errors:
+
+
+class ReccmpCsvFatalParserError(Exception):
+    """Cannot return even a single row from this document."""
+
+
+class CsvNoAddressError(ReccmpCsvFatalParserError):
+    """This file does not have an address attribute."""
+
+
+class CsvMultipleAddressError(ReccmpCsvFatalParserError):
+    """This file has more than one address attribute
+    (and it's not clear which one we should use)."""
```

Comment:
I'd generalise this error to a `CsvDuplicateColumnError` and also raise if another non-mandatory column is duplicated